api/src/main/scala/JobUtils.scala

```diff
@@ -0,0 +1,159 @@
+package org.wikiwatershed.mmw.geoprocessing
+
+import java.util.concurrent.atomic.LongAdder
+import collection.mutable
+import collection.concurrent.TrieMap
+import spray.json._
+import DefaultJsonProtocol._
+import geotrellis.proj4.{CRS, ConusAlbers, LatLng, WebMercator}
+import geotrellis.raster._
+import geotrellis.raster.rasterize._
+import geotrellis.raster.rasterize.Rasterizer.Options
+import geotrellis.raster.mapalgebra.focal.Kernel
+import geotrellis.vector._
+import geotrellis.vector.io._
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.s3._
+
+trait JobUtils {
+  val baseLayerReader = S3CollectionLayerReader("datahub-catalogs-us-east-1", "")
+
+  /**
+    * For an InputJson object, return a histogram of raster grouped count results.
+    *
+    * @param   input  The InputData
+    * @return         A histogram of results in JSON format
+    */
+  def getRasterGroupedCount(input: InputData) = {
+    val polygons = createMultipolygonFromInput(input)
+    val aoi = polygons.unionGeometries.asMultiPolygon.get
+    val rasterLayerIds = createRasterLayerIds(input.rasters, input.zoom)
+    val rasterLayers = rasterLayerIds.map { layer => fetchLayer(layer, aoi) }
+
+    Map("result" -> rasterGroupedCount(rasterLayers, aoi)).toJson
+  }
+
+  /**
+    * Given a zoom level from the input data, transform a list of raster filenames
+    * into a list of LayerIds.
+    *
+    * @param   rasterIds  A list of raster filenames
+    * @param   zoom       The input zoom level
+    * @return             A list of LayerIds
+    */
+  def createRasterLayerIds(rasterIds: List[String], zoom: Int) =
```

Comment:
Let's keep the main methods in a `Geoprocessing` trait, and move these helper methods to a new `Utils` trait.